cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/data/SparkTypeConverter.java

```diff
@@ -206,7 +206,7 @@ public Object toSparkDecimal(Object obj) {
             if( obj instanceof BigDecimal ) {
                 result = (BigDecimal) obj;
             } else if( obj instanceof Number ) {
-                result = new BigDecimal(((Number) obj).doubleValue());
+                result = BigDecimal.valueOf(((Number)obj).doubleValue());
```

Comment:
Because of floating point imprecision, you’re unlikely to get the value you expect from the BigDecimal(double) constructor.

From the JavaDocs:

    The results of this constructor can be somewhat unpredictable. One might assume that writing new BigDecimal(0.1) in Java creates a BigDecimal which is exactly equal to 0.1 (an unscaled value of 1, with a scale of 1), but it is actually equal to 0.1000000000000000055511151231257827021181583404541015625. This is because 0.1 cannot be represented exactly as a double (or, for that matter, as a binary fraction of any finite length). Thus, the value that is being passed in to the constructor is not exactly equal to 0.1, appearances notwithstanding.

Instead, you should use BigDecimal.valueOf, which uses a string under the covers to eliminate floating point rounding errors, or the constructor that takes a String argument.